<?xml version='1.0' encoding='UTF-8'?>
<root>
 <tabbed_widget parent="main_window" name="Main Window">
  <Tab tab_name="States" containers="1">
   <Container>
    <DockSplitter sizes="0.50055;0.49945" orientation="-" count="2">
     <DockSplitter sizes="0.5;0.5" orientation="|" count="2">
      <DockSplitter sizes="0.5;0.5" orientation="-" count="2">
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="-0.015569" left="1743095092.838370" top="0.612155"/>
         <limitY/>
         <curve color="#bcbd22" name="/mpc_solution/position/x"/>
         <curve color="#d62728" name="/mpc_solution/position/y"/>
         <curve color="#1f77b4" name="/mpc_solution/position/z"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="0.000253" left="1743095092.838370" top="0.015309"/>
         <limitY/>
         <curve color="#ff7f0e" name="/mpc_solution/linear_velocity/x"/>
         <curve color="#f14cc1" name="/mpc_solution/linear_velocity/y"/>
         <curve color="#9467bd" name="/mpc_solution/linear_velocity/z"/>
        </plot>
       </DockArea>
      </DockSplitter>
      <DockSplitter sizes="0.5;0.5" orientation="-" count="2">
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="-0.015578" left="1743095092.838370" top="0.612150"/>
         <limitY/>
         <curve color="#9467bd" name="/srbd_current/position/x"/>
         <curve color="#17becf" name="/srbd_current/position/y"/>
         <curve color="#bcbd22" name="/srbd_current/position/z"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="-0.002161" left="1743095092.838370" top="0.000912"/>
         <limitY/>
         <curve color="#1f77b4" name="/srbd_current/linear_velocity/x"/>
         <curve color="#d62728" name="/srbd_current/linear_velocity/y"/>
         <curve color="#1ac938" name="/srbd_current/linear_velocity/z"/>
        </plot>
       </DockArea>
      </DockSplitter>
     </DockSplitter>
     <DockSplitter sizes="0.502506;0.497494" orientation="|" count="2">
      <DockSplitter sizes="0.501104;0.498896" orientation="-" count="2">
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="-0.008881" left="1743095092.838370" top="-0.002198"/>
         <limitY/>
         <curve color="#1ac938" name="/mpc_solution/orientation/x"/>
         <curve color="#ff7f0e" name="/mpc_solution/orientation/y"/>
         <curve color="#f14cc1" name="/mpc_solution/orientation/z"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="0.000041" left="1743095092.838370" top="0.101065"/>
         <limitY/>
         <curve color="#d62728" name="/mpc_solution/angular_velocity/x"/>
         <curve color="#1ac938" name="/mpc_solution/angular_velocity/y"/>
         <curve color="#ff7f0e" name="/mpc_solution/angular_velocity/z"/>
        </plot>
       </DockArea>
      </DockSplitter>
      <DockSplitter sizes="0.501104;0.498896" orientation="-" count="2">
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="-0.008796" left="1743095092.838370" top="-0.002176"/>
         <limitY/>
         <curve color="#17becf" name="/srbd_current/orientation/x"/>
         <curve color="#bcbd22" name="/srbd_current/orientation/y"/>
         <curve color="#1f77b4" name="/srbd_current/orientation/z"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="-0.031540" left="1743095092.838370" top="0.023643"/>
         <limitY/>
         <curve color="#ff7f0e" name="/srbd_current/angular_velocity/x"/>
         <curve color="#f14cc1" name="/srbd_current/angular_velocity/y"/>
         <curve color="#9467bd" name="/srbd_current/angular_velocity/z"/>
        </plot>
       </DockArea>
      </DockSplitter>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="Contact Forces" containers="1">
   <Container>
    <DockSplitter sizes="1" orientation="-" count="1">
     <DockSplitter sizes="0.5;0.5" orientation="|" count="2">
      <DockSplitter sizes="0.50055;0.49945" orientation="-" count="2">
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="-2.622508" left="1743095092.838370" top="114.256962"/>
         <limitY/>
         <curve color="#f14cc1" name="/mpc_solution/contacts[0]/force/x"/>
         <curve color="#9467bd" name="/mpc_solution/contacts[0]/force/y"/>
         <curve color="#17becf" name="/mpc_solution/contacts[0]/force/z"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="-2.719480" left="1743095092.838370" top="119.074473"/>
         <limitY/>
         <curve color="#d62728" name="/mpc_solution/contacts[2]/force/x"/>
         <curve color="#1ac938" name="/mpc_solution/contacts[2]/force/y"/>
         <curve color="#ff7f0e" name="/mpc_solution/contacts[2]/force/z"/>
        </plot>
       </DockArea>
      </DockSplitter>
      <DockSplitter sizes="0.50055;0.49945" orientation="-" count="2">
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="-1.825124" left="1743095092.838370" top="81.264441"/>
         <limitY/>
         <curve color="#17becf" name="/mpc_solution/contacts[1]/force/x"/>
         <curve color="#bcbd22" name="/mpc_solution/contacts[1]/force/y"/>
         <curve color="#1f77b4" name="/mpc_solution/contacts[1]/force/z"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="-1.710732" left="1743095092.838370" top="81.715448"/>
         <limitY/>
         <curve color="#f14cc1" name="/mpc_solution/contacts[3]/force/x"/>
         <curve color="#9467bd" name="/mpc_solution/contacts[3]/force/y"/>
         <curve color="#17becf" name="/mpc_solution/contacts[3]/force/z"/>
        </plot>
       </DockArea>
      </DockSplitter>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="Contact Positions" containers="1">
   <Container>
    <DockSplitter sizes="1" orientation="-" count="1">
     <DockSplitter sizes="0.5;0.5" orientation="|" count="2">
      <DockSplitter sizes="0.50055;0.49945" orientation="-" count="2">
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="-0.033903" left="1743095092.838370" top="0.121864"/>
         <limitY/>
         <curve color="#bcbd22" name="/srbd_current/contacts[0]/position/x"/>
         <curve color="#1f77b4" name="/srbd_current/contacts[0]/position/y"/>
         <curve color="#d62728" name="/srbd_current/contacts[0]/position/z"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="-0.121366" left="1743095092.838370" top="-0.023327"/>
         <limitY/>
         <curve color="#9467bd" name="/srbd_current/contacts[2]/position/x"/>
         <curve color="#17becf" name="/srbd_current/contacts[2]/position/y"/>
         <curve color="#bcbd22" name="/srbd_current/contacts[2]/position/z"/>
        </plot>
       </DockArea>
      </DockSplitter>
      <DockSplitter sizes="0.50055;0.49945" orientation="-" count="2">
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="-0.030128" left="1743095092.838370" top="0.144719"/>
         <limitY/>
         <curve color="#1ac938" name="/srbd_current/contacts[1]/position/x"/>
         <curve color="#ff7f0e" name="/srbd_current/contacts[1]/position/y"/>
         <curve color="#f14cc1" name="/srbd_current/contacts[1]/position/z"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="-0.125392" left="1743095092.838370" top="0.147007"/>
         <limitY/>
         <curve color="#1f77b4" name="/srbd_current/contacts[3]/position/x"/>
         <curve color="#d62728" name="/srbd_current/contacts[3]/position/y"/>
         <curve color="#1ac938" name="/srbd_current/contacts[3]/position/z"/>
        </plot>
       </DockArea>
      </DockSplitter>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="Contact Active" containers="1">
   <Container>
    <DockSplitter sizes="1" orientation="-" count="1">
     <DockSplitter sizes="0.5;0.5" orientation="|" count="2">
      <DockSplitter sizes="0.50055;0.49945" orientation="-" count="2">
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="0.900000" left="1743095092.838370" top="1.100000"/>
         <limitY/>
         <curve color="#d62728" name="/mpc_solution/contacts[0]/active"/>
         <curve color="#1ac938" name="/mpc_solution/contacts[1]/active"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="0.900000" left="1743095092.838370" top="1.100000"/>
         <limitY/>
         <curve color="#1f77b4" name="/mpc_solution/contacts[2]/active"/>
         <curve color="#d62728" name="/mpc_solution/contacts[3]/active"/>
        </plot>
       </DockArea>
      </DockSplitter>
      <DockSplitter sizes="0.50055;0.49945" orientation="-" count="2">
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="-0.100000" left="1743095092.838370" top="0.100000"/>
         <limitY/>
         <curve color="#1ac938" name="/srbd_current/contacts[0]/active"/>
         <curve color="#ff7f0e" name="/srbd_current/contacts[1]/active"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
         <range right="1743095097.836562" bottom="-0.100000" left="1743095092.838370" top="0.100000"/>
         <limitY/>
         <curve color="#f14cc1" name="/srbd_current/contacts[2]/active"/>
         <curve color="#9467bd" name="/srbd_current/contacts[3]/active"/>
        </plot>
       </DockArea>
      </DockSplitter>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="XY CoM" containers="1">
   <Container>
    <DockSplitter sizes="1" orientation="-" count="1">
     <DockArea name="...">
      <plot style="Lines" mode="XYPlot" flip_x="false" flip_y="false">
       <range right="0.049065" bottom="-0.000637" left="0.048659" top="-0.000417"/>
       <limitY/>
       <curve curve_x="/mpc_solution/position/x" curve_y="/mpc_solution/position/y" color="#1f77b4" name="/mpc_solution/position/[x;y]"/>
       <curve curve_x="/srbd_current/position/x" curve_y="/srbd_current/position/y" color="#9467bd" name="/srbd_current/position/[x;y]"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="Solving Times" containers="1">
   <Container>
    <DockSplitter sizes="1" orientation="-" count="1">
     <DockArea name="...">
      <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
       <range right="1743095097.836562" bottom="-0.140477" left="1743095092.838370" top="5.789393"/>
       <limitY/>
       <curve color="#17becf" name="/mpc_statistics/full/statistics[4]/value"/>
       <curve color="#bcbd22" name="/wbid_statistics/full/statistics[0]/value"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <currentTabIndex index="0"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="0"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <parameters delimiter="0" time_axis=""/>
  </plugin>
  <plugin ID="DataLoad MCAP"/>
  <plugin ID="DataLoad ROS bags">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
   <selected_topics value=""/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="ROS Topic Subscriber">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
   <selected_topics value="/mpc_solution;/srbd_current"/>
  </plugin>
  <plugin ID="UDP Server"/>
  <plugin ID="WebSocket Server"/>
  <plugin ID="ZMQ Subscriber"/>
  <plugin ID="Fast Fourier Transform"/>
  <plugin ID="Quaternion to RPY"/>
  <plugin ID="Reactive Script Editor">
   <library code="--[[ Helper function to create a series from arrays&#xa;&#xa; new_series: a series previously created with ScatterXY.new(name)&#xa; prefix:     prefix of the timeseries, before the index of the array&#xa; suffix_X:   suffix to complete the name of the series containing the X value. If [nil], use the index of the array.&#xa; suffix_Y:   suffix to complete the name of the series containing the Y value&#xa; timestamp:   usually the tracker_time variable&#xa;              &#xa; Example:&#xa; &#xa; Assuming we have multiple series in the form:&#xa; &#xa;   /trajectory/node.{X}/position/x&#xa;   /trajectory/node.{X}/position/y&#xa;   &#xa; where {N} is the index of the array (integer). We can create a reactive series from the array with:&#xa; &#xa;   new_series = ScatterXY.new(&quot;my_trajectory&quot;) &#xa;   CreateSeriesFromArray( new_series, &quot;/trajectory/node&quot;, &quot;position/x&quot;, &quot;position/y&quot;, tracker_time );&#xa;--]]&#xa;&#xa;function CreateSeriesFromArray( new_series, prefix, suffix_X, suffix_Y, timestamp )&#xa;  &#xa;  --- clear previous values&#xa;  new_series:clear()&#xa;  &#xa;  --- Append points to new_series&#xa;  index = 0&#xa;  while(true) do&#xa;&#xa;    x = index;&#xa;    -- if not nil, get the X coordinate from a series&#xa;    if suffix_X ~= nil then &#xa;      series_x = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_X) )&#xa;      if series_x == nil then break end&#xa;      x = series_x:atTime(timestamp)&#x9; &#xa;    end&#xa;    &#xa;    series_y = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_Y) )&#xa;    if series_y == nil then break end &#xa;    y = series_y:atTime(timestamp)&#xa;    &#xa;    new_series:push_back(x,y)&#xa;    index = index+1&#xa;  end&#xa;end&#xa;&#xa;--[[ Similar to the built-in function GetSeriesNames(), but select only the names with a give prefix. --]]&#xa;&#xa;function GetSeriesNamesByPrefix(prefix)&#xa;  -- GetSeriesNames(9 is a built-in function&#xa;  all_names = GetSeriesNames()&#xa;  filtered_names = {}&#xa;  for i, name in ipairs(all_names)  do&#xa;    -- check the prefix&#xa;    if name:find(prefix, 1, #prefix) then&#xa;      table.insert(filtered_names, name);&#xa;    end&#xa;  end&#xa;  return filtered_names&#xa;end&#xa;&#xa;--[[ Modify an existing series, applying offsets to all their X and Y values&#xa;&#xa; series: an existing timeseries, obtained with TimeseriesView.find(name)&#xa; delta_x: offset to apply to each x value&#xa; delta_y: offset to apply to each y value &#xa;  &#xa;--]]&#xa;&#xa;function ApplyOffsetInPlace(series, delta_x, delta_y)&#xa;  -- use C++ indeces, not Lua indeces&#xa;  for index=0, series:size()-1 do&#xa;    x,y = series:at(index)&#xa;    series:set(index, x + delta_x, y + delta_y)&#xa;  end&#xa;end&#xa;"/>
   <scripts/>
  </plugin>
  <plugin ID="CSV Exporter"/>
  <plugin ID="ROS /rosout Visualization"/>
  <plugin ID="ROS Topic Re-Publisher"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles/>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations/>
 <snippets/>
 <!-- - - - - - - - - - - - - - - -->
</root>

